src/as3hx/Writer.hx

```diff
@@ -1572,6 +1572,7 @@ class Writer
                     if (inits.empty() || conds.empty())
                         return false;
  
+                    if (conds[0].match(EBinop("&&" | "||", _, _, _))) return false;
```

Comment:
This check seems quite weak... Wouldn't it be better to invert it to something like `if(!isSimpleExpression)` to make sure we go for the `while` by default?